control-plane/connect-inject/consul_dataplane_sidecar.go

```diff
@@ -190,6 +190,60 @@ func (w *MeshWebhook) getContainerSidecarCommand(namespace corev1.Namespace, mpi
 		cmd = append(cmd, fmt.Sprintf("-envoy-admin-bind-port=%d", 19000+mpi.serviceIndex))
 	}
 
+	metricsServer, err := w.MetricsConfig.shouldRunMergedMetricsServer(pod)
+	if err != nil {
+		return nil, fmt.Errorf("unable to determine if merged metrics is enabled: %w", err)
+	}
+	if metricsServer {
+		prometheusScrapePath := w.MetricsConfig.prometheusScrapePath(pod)
+		mergedMetricsPort, err := w.MetricsConfig.mergedMetricsPort(pod)
+		if err != nil {
+			return nil, fmt.Errorf("unable to determine if merged metrics port: %w", err)
+		}
+		cmd = append(cmd, "-telemetry-prom-scrape-path="+prometheusScrapePath,
+			"-telemetry-prom-merge-port="+mergedMetricsPort)
+
+		serviceMetricsPath := pod.Annotations[annotationServiceMetricsPath]
+		serviceMetricsPort := pod.Annotations[annotationServiceMetricsPort]
```

Comment:
I think we should use the `w.MetricsConfig.serviceMetricsPath` and `w.MetricsConfig.serviceMetricsPort`  so that we get the same defaults handling.

Reply:
Nice catch and fixed a few other things!